Reader Post: Add animation to train of likes

dvdchr opened the pull request:
Refs: `pctCYC-8w-p2`

## Summary

This introduces interactivity to the likes view component in the Reader Detail screen. When liking a post, the avatar of the current user is appended at the trailing position. Furthermore, there can be maximum 6 avatars displayed, where up to 5 other bloggers may be displayed, and lastly the current user's avatar is displayed at the end _if_ the user liked the post.

Here's a summary of the code changes:
- Modified the data flow to pass in `[String]` (representing avatar URL strings) instead of `[LikeUser]` objects from the `ReaderDetailCoordinator` instance down to the `ReaderDetailLikesView`.
  - Since the reader post will be interactive (reacting to Like toolbar button taps), it's much easier to pass avatar URL by fetching the value from `WPAccount`, instead of having to create  an unmanaged `LikeUser` instances whenever the user taps the Like button.
- Added a new method in the `ReaderDetailView` protocol: `updateSelfLike(with avatarURLString: String)`, to specifically handle singular like addition / subtraction.
  - The current `updateLikes` method is set up as a one-time setup, while the `updateSelfLike` may be called multiple times while the Reader Post is opened.
- The number of total likes passed from the coordinator to the view is intentionally reduced by 1 if the post is liked by current user. This will be handled by the `ReaderDetailLikesView` where an extra like will be added back if the post is liked.
  - Note that this is a set up to update the summary label text (for adding "You and..." part). Ref: `pbArwn-2mK-p2#comment-3436`
- I'll defer adding the unit tests to the upcoming PR when addressing `pbArwn-2mK-p2#comment-3436`.

Here's a preview of the animation:
  
https://user-images.githubusercontent.com/1299411/127351718-d68a102b-9f61-4157-a1b5-0fdd577e393c.mov
 
Lastly, I _think_ there's a potential race condition when adding/removing likes from a post – which causes a post to have `isLiked == false`, but a `LikeUser` object representing the current user exists in the database. However, I'll investigate this later as this is a separate issue from this PR (and will open an issue later with the repro steps).

## To test

- Open a post with 0 likes. 
  - Verify the likes view is shown when the post is liked.
  - Verify the likes view is hidden when the post is unliked.
- Open a post where you're the only one liking the post.
  - Verify the likes view is hidden when the post is unliked.
  - Verify the likes view is shown when the post is liked.
- Open a post with more than 5 likes from other people. 
  - Verify that it shows 6 avatars when the post is liked.
  - Verify that it shows 5 avatars when the post is unliked.
- Verify that the animation works fine when the post is liked and unliked.
- Verify that the animation direction is correct in RTL language.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
The change is only scoped to Reader Post.

3. What automated tests I added (or what prevented me from doing so)
Will be added in the upcoming PR that addresses `pbArwn-2mK-p2#comment-3436`.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

dvdchr committed:
Add AccountService as dependency for ReaderDetailCoordinator

dvdchr committed:
Modify updateLikes method to accept String instead of LikeUser instances

This change will help simplify the implementation later on when reacting to like button actions from ReaderDetailToolbar. Previously, if the method requires LikeUser, then we'd have to unnecessarily create `LikeUser` objects based on `WPAccount`.

If the method accepts URL strings instead, we can just simply pass in the value of `avatarUrl` from `WPAccount` without converting it to a `LikeUser` instance.

dvdchr committed:
Add capability to show current user's avatar on the right-most position

This also adds a new method, `updateSelfLike`, to later be used for reacting to like button taps from the toolbar.

dvdchr committed:
React to likeButton tap from ReaderDetailToolbar

dvdchr committed:
Add animation when the likes view is visible, plus RTL support

dvdchr committed:
Remove unused comments

dvdchr committed:
Fix comments to explain the correct behavior

ScoutHarris reviewed (approved):
This is nice!!

:shipit:

ScoutHarris commented:
Also, just to be clear.

> I'll defer adding the unit tests to the upcoming PR

I'm not sure if this is referring to new or existing tests, but please be sure the existing tests pass before merging. Thanks!!

dvdchr committed:
Fix tests

dvdchr commented:
> please be sure the existing tests pass before merging

Yep. I meant for the new tests, but the existing tests broke because I added a new method for `ReaderDetailView` protocol. It's now fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/16936/commits/4365783eb1bc2543034d3377d427496f4be584cb 😄
